utilities/src/main/java/org/craftercms/commons/monitoring/rest/MonitoringRestControllerBase.java

```diff
@@ -36,18 +41,34 @@
     public final static String VERSION_URL = "/version";
 
     @GetMapping(ROOT_URL + MEMORY_URL)
-    public MemoryInfo getCurrentMemory() {
-        return MemoryInfo.getCurrentMemory();
+    public MemoryInfo getCurrentMemory(@RequestParam(name = "token", required = true) String token)
+            throws InvalidMonitoringTokenException {
+        if (StringUtils.isNotEmpty(token) && StringUtils.equals(token, getConfiguredToken())) {
+            return MemoryInfo.getCurrentMemory();
+        } else {
+            throw new InvalidMonitoringTokenException("Invalid token for monitoring authorization");
+        }
     }
 
     @GetMapping(ROOT_URL + STATUS_URL)
-    public StatusInfo getCurrentStatus() {
-        return StatusInfo.getCurrentStatus();
+    public StatusInfo getCurrentStatus(@RequestParam(name = "token", required = true) String token)
+            throws InvalidMonitoringTokenException {
+        if (StringUtils.isNotEmpty(token) && StringUtils.equals(token, getConfiguredToken())) {
+            return StatusInfo.getCurrentStatus();
+        } else {
+            throw new InvalidMonitoringTokenException("Invalid token for monitoring authorization");
```

Comment:
Change the message of the exception to `Monitoring authorization failed, invalid token`.